src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -111,11 +131,15 @@ private static Long getTheme(String widget) {
             try {
                 if (!valid) {
                     // Close old themes.
-                    for (Long value : widgetToTheme.values()) {
-                        closeTheme(value);
+                    if (!dpiAwareWidgetToTheme.isEmpty()) {
+                        for (Long value :
+                             dpiAwareWidgetToTheme.get(dpi).values()) {
+                            closeTheme(value);
+                        }
+                        dpiAwareWidgetToTheme.get(dpi).clear();
+                        dpiAwareWidgetToTheme.clear();
```

Comment:
You clear the entire map, yet it don't call `closeTheme` for dpi values other than the passed one.

You should iterate over `dpiAwareWidgetToTheme.values().values()` (pseudo-code).